scheds/rust/scx_rustland/src/topology.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Andrea Righi <andrea.righi@canonical.com>
+
+// This software may be used and distributed according to the terms of the
+// GNU General Public License version 2.
+
+use std::collections::HashMap;
+use std::fs;
+
+/// scx_rustland: CPU topology.
```

Comment:
Not necessary for this change, but his could be useful to provide as a rust crate as well. In scx_rusty we collect the host's topology to determine what domains we should create for the scheduler. If we had a single utility for mapping logical CPUs -> physical cores -> LLC packages -> NUMA nodes, I think that would be quite handy.

Reply:
@Decave yes, there's a lot of generic stuff that we can reuse in other schedulers (the cpu topology, the allocator, the bpf abstraction). It would be really nice to have them in the scx_utils crate. I'll start look at that and see if we can migrate some of these "generic building blocks" in there.